modules/library/coverage/src/test/java/org/geotools/image/ImageWorkerTest.java

```diff
@@ -1713,6 +1720,40 @@ public void testBandMerge() {
         assertNoData(image, null);
     }
 
+    @Test
+    public void testBandMergeWithNodata() throws FileNotFoundException, IOException {
+        double noDataValue = -10000d;
+        Range nodata = RangeFactory.create(noDataValue, noDataValue);
+        double[] background = new double[] {noDataValue};
+
+        RenderedImage twoBands = null;
+        PlanarImage pi = PlanarImage.wrapRenderedImage(imageWithNodata2);
+        pi.setProperty("GC_NODATA", new NoDataContainer(nodata));
+
+        ImageWorker worker =
+                new ImageWorker()
+                        .setNoData(nodata)
+                        .setBackground(background)
+                        .mosaic(
+                                new RenderedImage[] {pi},
+                                MosaicDescriptor.MOSAIC_TYPE_OVERLAY,
+                                null,
+                                null,
+                                new double[][] {{-32768}},
+                                null);
+        twoBands = worker.addBand(worker.getRenderedImage(), false).getRenderedImage();
+        RenderedImage singleBand = new ImageWorker(twoBands).retainBands(1).getRenderedImage();
+        RenderedImage threeBands =
+                new ImageWorker(twoBands).addBand(singleBand, false).getRenderedImage();
+
+        // Check that the noData holes are still noData after bandMerge
```

Comment:
What happens if, by accident, the different input bands had different nodata?

Reply:
I think that at the moment it isn't supported since there is only a single nodata value support, not an array. Do we ever had a case with different nodata in input?

Reply:
Not really, just thinking out loud